File: backend/flowlet/src/main/java/com/example/flowlet/account/domain/model/Account.java
package com.example.flowlet.account.domain.model;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record Account(
    Long accountId,
    String providerName,
    String accountName,
    AccountCategory accountCategory,
    BalanceSide balanceSide,
    BigDecimal initialBalance,
    boolean active,
    Integer displayOrder,
    LocalDateTime createdAt,
    LocalDateTime updatedAt
) {
}

File: frontend/src/features/account/types/account.ts
export type AccountType = 'CHECKING' | 'SAVINGS' | 'OTHER'

export type Account = {
  accountId: number
  bankName: string
  accountName: string
  accountType: AccountType
  active: boolean
  createdAt: string
  updatedAt: string
}

export type CreateAccountInput = {
  bankName: string
  accountName: string
  accountType: AccountType
  active: boolean
}

export const accountTypeLabels: Record<AccountType, string> = {
  CHECKING: '普通預金',
  SAVINGS: '貯蓄預金',
  OTHER: 'その他',
}
